lgc/interface/lgc/LgcDialect.td

```diff
@@ -79,3 +79,88 @@ def BufferPtrDiffOp : LgcOp<"buffer.ptr.diff", [Memory<[]>, WillReturn]> {
     If `lhs` and `rhs` are both null, the result is 0.
   }];
 }
+
+def GenericLocationOp : OpClass<LgcDialect> {
+  let arguments = (ins AttrI1:$perPrimitive, AttrI32:$location, I32:$locOffset, I32:$elemIdx, I32:$arrayIndex);
+
+  let summary = "family of operations that reference generic shader I/O locations";
+  let description = [{
+    Generic (graphics) shader I/O interfaces are addressed in terms of locations. Locations are 4-element vectors with
+    32-bit elements. Within an interface, the location vectors are indexed by `location + locOffset`. The elements of
+    the vector are indexed by `elemIdx`.
+
+    For some shader stages, the interface is inherently an array. For example, GS inputs are an array of per-vertex
+    inputs, where the size of the array depends on the input primitive type (1 for points, 2 for lines, etc.).
+
+    Some shader stages have both array and non-array interfaces. Specifically, TCS outputs and TES inputs have per-patch
+    locations (non-array) and per-control-point locations (array with one entry per control point).
+
+    For array interfaces, `arrayIndex` is used to index into the array. For non-array interfaces, `arrayIndex` is
```

Comment:
Does the array index stand for vertex array or primitive array in GS/Mesh shader/TCS/TES. If so, maybe explain it a little concretely in the comments.

Reply:
Yes, I'm going to try to make it clearer.